daemon/daemon_unix.go

```diff
@@ -248,12 +248,11 @@ func (daemon *Daemon) adaptContainerSettings(hostConfig *containertypes.HostConf
 		hostConfig.ShmSize = container.DefaultSHMSize
 	}
 	var err error
-	if hostConfig.SecurityOpt == nil {
-		hostConfig.SecurityOpt, err = daemon.generateSecurityOpt(hostConfig.IpcMode, hostConfig.PidMode, hostConfig.Privileged)
-		if err != nil {
-			return err
-		}
+	opts, err := daemon.generateSecurityOpt(hostConfig.IpcMode, hostConfig.PidMode, hostConfig.Privileged)
```

Comment:
This looks like your container would be joining an existing container,  perhaps kubernetes is joining the pid container of the PAUSE and the pause container is running as container_t:s0:C1,c2.

The problem might be kubernetes not passing the security options at the time it is creating the pause container?